etl/steps/__init__.py

```diff
@@ -574,3 +584,29 @@ def run(self) -> None:
 
     def checksum_output(self) -> str:
         return get_etag(f"https://{self.path}")
+
+
+class DataStepPrivate(DataStep):
+    is_public = False
+
+    @staticmethod
+    def _make_dataset_private(dest_dir: str) -> None:
+        ds = catalog.Dataset(dest_dir)
+        ds.metadata.is_public = False
+        ds.save()
+
+    def _run_py(self) -> None:
+        super()._run_py()
+        self._make_dataset_private(self._dest_dir.as_posix())
+
+    def _run_notebook(self) -> None:
+        super()._run_notebook()
+        self._make_dataset_private(self._dest_dir.as_posix())
```

Comment:
Not important now, but if we added more `_run` methods, we'd probably just want an `after_run()` hook you could define in the subclass.

Reply:
That's much better, fixed it